src/main/java/io/airlift/slice/SizeOf.java

```diff
@@ -97,6 +105,26 @@ public static long sizeOf(Object[] array)
         return (array == null) ? 0 : sizeOfObjectArray(array.length);
     }
 
+    public static <T> long sizeOf(Optional<T> optional, ToLongFunction<T> valueSize)
+    {
+        return optional.isPresent() ? OPTIONAL_INSTANCE_SIZE + valueSize.applyAsLong(optional.get()) : 0;
```

Comment:
```suggestion
        return optional.map(value -> OPTIONAL_INSTANCE_SIZE + valueSize.applyAsLong(value)).orElse(0);
```

Reply:
I was trying to avoid using capturing lambdas for efficiency reasons as this is a low level library method that may in theory be used in performance critical places.